File: Android Basics/MusicStation/app/src/main/java/com/example/alifyzfpires/musicstation/MainActivity.java
package com.example.alifyzfpires.musicstation;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView reproduzir = (TextView) findViewById(R.id.reproduzir);
        TextView playlist   = (TextView) findViewById(R.id.playlist);
        TextView premium    = (TextView) findViewById(R.id.premium);
        TextView config     = (TextView) findViewById(R.id.configurações);

        reproduzir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent reproduzir = new Intent(MainActivity.this, Reproduzir.class);
                startActivity(reproduzir);
            }
        });

        playlist.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent playlist = new Intent(MainActivity.this, Playlist.class);
                startActivity(playlist);
            }
        });

        premium.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent premium = new Intent(MainActivity.this, Premium.class);
                startActivity(premium);
            }
        });

        config.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent configuracoes = new Intent(MainActivity.this, Configuracoes.class);
                startActivity(configuracoes);
            }
        });


    }
}

File: NotesApp/app/src/main/java/com/alifyz/notesapp/MainActivity.kt
package com.alifyz.notesapp

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.Observer
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.helper.ItemTouchHelper
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author Alifyz Pires
 * @date 01/20/2019
 */
class MainActivity : AppCompatActivity() {

    //Instância do Banco de Dados que iremos usar
    lateinit var database : NotesDatabase

    //Nosso Adapter customizado para exibição das anotações
    val adapter = Adapter()


    lateinit var notesList : LiveData<List<NotesEntity>>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupDatabase()
        setupRecyclerView()

        notesList.observe(this, Observer {
            it?.let { lista -> adapter.setData(lista) }
        })

        fab_new_note.setOnClickListener {
            val newNoteActivity = Intent(this, NewNoteActivity::class.java)
            startActivity(newNoteActivity)
        }
    }

    //Criação do Banco de Dados
    fun setupDatabase() {
        database = NotesDatabase.getInstance(this)
        notesList = database.DAO().getAll()
    }

    //Configuração do RecyclerView
    fun setupRecyclerView() {
        val touchHelper = ItemTouchHelper(TouchHelper(adapter, this))
        touchHelper.attachToRecyclerView(recyclerview)
        recyclerview.layoutManager = LinearLayoutManager(this)
        recyclerview.adapter = adapter
    }

}
